freq/views.py

```diff
@@ -1,283 +1,913 @@
+        }
+
+        self.request.session['startpage']['timeseries_length'] = len(data)
+        self.request.session.modified = True
+
+        return self.data
 
-    def flush(self, request):
-        request.session.update(self.default)
-        del(request.session['disabled'])
-        request.session['disabled'] = self.disabled(request)
+
+    @cached_property
+    def error_message(self):
+        if len(self.timeseries['values']) == 0:
+            return 'No time series found for this location, please select ' \
+                   'another.'
+        return ''
+
+    # ------------------------------------------------------------------------ #
+    ### Date handling
+    JS_EPOCH = dt.datetime(1970, 1, 1)
+
+    def datetime_to_js(self, date_time):
+        if date_time is not None:
+            return int((date_time - self.JS_EPOCH).total_seconds() * 1000)
+
+    def js_to_datetime(self, date_time):
+        if date_time is not None:
+            return self.JS_EPOCH + dt.timedelta(seconds=date_time/1000)
+
+    @cached_property
+    def today(self):
+        return today()
+
+    # TODO: 'old' code once written for the map part, probably obsolete.
+    # @cached_property
+    # def current(self):
+    #     return self.active if self.active else 'map_'
+    #
+    # @cached_property
+    # def start(self):
+    #     start_old = self.request.session[self.current].get('start', self.today)
+    #     return self.request.GET.get('start', start_old)
+    #
+    # @cached_property
+    # def end(self):
+    #     start_old = self.request.session[self.current].get('end', self.today)
+    #     return self.request.GET.get('end', start_old)
+
+    @property
+    def time_window(self):
+        start = self.datetime_to_js(dt.datetime.strptime(
+            self.request.session['startpage']['datepicker']['start'], '%d-%m-%Y'
+        ))
+        end = self.datetime_to_js(dt.datetime.strptime(
+            self.request.session['startpage']['datepicker']['end'], '%d-%m-%Y'
+        ))
+        return {
+            'start': start,
+            'end': end
+        }
+
+    # ------------------------------------------------------------------------ #
+    ### Button handling:
+
+    @cached_property
+    def length(self):
+        length = self.request.session['startpage']['timeseries_length']
+        if length < 0:
+            return len(self.timeseries['values'])
+        return length
+
+    @cached_property
+    def spinner_value(self):
+        return self.request.session[self.active]['spinner_0']['value']
+
+    @cached_property
+    def dropdown_selected(self):
+        return self.request.session[self.active]['dropdown_0']['value']
+
+    @cached_property
+    def dropdown_unselected(self):
+        return [x for x in self.dropdown_options if x != self.dropdown_selected]
+
+    @cached_property
+    def datepicker_start(self):
+        return [int(x) for x in self.request.session['startpage']['datepicker']['start'].split('-')]
```

Comment:
Gelieve HEEL zeker te weten dat `int(x)` nooit fout gaat want anders krijg je her en der moeilijk de debuggen fouten. Oftwel de "start" moet gegarandeerd in het juiste formaat staan. Altijd.